examples/api/lib/widgets/animated_list/animated_list_separated.0.dart

```diff
@@ -0,0 +1,274 @@
+void main() {
+  runApp(const AnimatedListSeparatedSample());
+}
+
+class AnimatedListSeparatedSample extends StatefulWidget {
+  const AnimatedListSeparatedSample({super.key});
+
+  @override
+  State<AnimatedListSeparatedSample> createState() => _AnimatedListSeparatedSampleState();
+}
+
+class _AnimatedListSeparatedSampleState extends State<AnimatedListSeparatedSample> {
+  final GlobalKey<AnimatedListSeparatedState> _listKey = GlobalKey<AnimatedListSeparatedState>();
+  late ListModel<int> _list;
+  int? _selectedItem;
+  late int _nextItem; // The next item inserted when the user presses the '+' button.
+
+  @override
+  void initState() {
+    super.initState();
+    _list = ListModel<int>(
+      listKey: _listKey,
+      initialItems: <int>[0, 1, 2],
+      removedItemBuilder: _buildRemovedItem,
+      removedSeparatorBuilder: _buildRemovedSeparator,
+    );
+    _nextItem = 3;
+  }
+
+  // Used to build list items that haven't been removed.
+  Widget _buildItem(BuildContext context, int index, Animation<double> animation) {
+    return CardItem(
+      animation: animation,
+      item: _list[index],
+      selected: _selectedItem == _list[index],
+      onTap: () {
+        setState(() {
+          _selectedItem = _selectedItem == _list[index] ? null : _list[index];
+        });
+      },
+    );
+  }
+
+  // Used to build separators that haven't been removed.
+  Widget _buildSeparator(BuildContext context, int index, Animation<double> animation) {
+    return ItemSeparator(
+      animation: animation,
+      item: _list[index],
+    );
+  }
+
+  /// The builder function used to build items that have been removed.
+  ///
+  /// Used to build an item after it has been removed from the list. This method
+  /// is needed because a removed item remains visible until its animation has
+  /// completed (even though it's gone as far as this ListModel is concerned).
+  /// The widget will be used by the [AnimatedListSeparatedState.removeSeparatedItem] method's
+  /// `itemBuilder` parameter.
+  Widget _buildRemovedItem(int item, BuildContext context, Animation<double> animation) {
+    return CardItem(
+      animation: animation,
+      item: item,
+      // No gesture detector here: we don't want removed items to be interactive.
+    );
+  }
+
+  // Build a separator for items that have been removed from the list.
+  /// The widget will be used by the [AnimatedListSeparatedState.removeSeparatedItem] method's
+  /// `separatorBuilder` parameter.
+  Widget _buildRemovedSeparator(int item , BuildContext context, Animation<double> animation) => SizeTransition(
+                sizeFactor: animation,
+                child: ItemSeparator(
+                  animation: animation,
+                  item: item,
+                )
+              );
+
+  // Insert the "next item" into the list model.
+  void _insert() {
+    final int index = _selectedItem == null ? _list.length : _list.indexOf(_selectedItem!);
+    _list.insert(index, _nextItem++);
```

Comment:
to make this code clearer to new programmers, i recommend increasing _nextItem separate from inserting it into the list, so it's 100% clear whether the inserted value is the pre-increment or post-increment value.